axum-extra/src/protobuf.rs

```diff
@@ -0,0 +1,303 @@
+//! Protocol Buffer extractor and response.
+
+use axum::{
+    async_trait,
+    body::{Bytes, HttpBody},
+    extract::{rejection::BytesRejection, FromRequest, RequestParts},
+    response::{IntoResponse, Response},
+    BoxError,
+};
+use http::StatusCode;
+use prost::Message;
+use std::ops::{Deref, DerefMut};
+
+/// Protocol Buffer extractor and response.
+///
+/// When used as an extractor, it can decode request bodies into some type that
+/// implements [`prost::Message`]. The request will be rejected (and a [`ProtoBufRejection`] will
+/// be returned) if:
```

Comment:
I think we should put this into the extractor section below.

Reply:
I updated the documentation and adapted the documentation style from the [JsonLines](https://docs.rs/axum-extra/latest/axum_extra/json_lines/struct.JsonLines.html) extractor.